<!DOCTYPE html>
<html lang="ja">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>エラーの種類によって例外処理に条件分岐を持たせる</title>
  </head>
  <body>
    <script>
      try {
        // 数値に対して文字列のメソッドであるtoUpperCaseを呼んだ場合
        const num = 1;
        num.toUpperCase();

        // 存在しない変数val2を参照した場合
        let val1 = val2 + 1;
      } catch (error) {
        // 例外の処理に応じて処理を分岐する
        if (error instanceof TypeError) {
          console.log("Process when TypeError occurs");
        } else if (error instanceof ReferenceError) {
          console.log("Process when ReferenceError occurs");
        }
      }
    </script>
  </body>
</html>
